### PR TITLE
Improve ENR updates

### PR DESCRIPTION
**Pull request by AgeManning**
## Issue Addressed

I've noticed that on boot, we are aggressively updating the ENR. This increases our sequence number and makes for more ENRs to propagate though the DHT. 

After looking into it, it seems we are updating the ENR unnecessarily and also could be updating the wrong fields. When we listen to an ipv6 address we would update the ipv4 tcp port on the ENR. 

## Proposed Changes

This PR adjusts some logic so that we only update the ENR if there is a change, and also we update the ip4/ip6 fields respectively.

### CODE REVIEW

File: beacon_node/lighthouse_network/src/discovery/mod.rs [hunk 1]
```diff
@@ -398,33 +398,70 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
...
+                return Ok(false);
+            }
+            "tcp6"
+            } else {
+            if self.discv5.external_enr().read().tcp4() == Some(port) {
+                // The field is already set to the same value, nothing to do
+                return Ok(false);
+            }
+            "tcp"
+        };
```
Comment (dapplion) [inline beacon_node/lighthouse_network/src/discovery/mod.rs:416]: This needs `cargo fmt`

Comment (AgeManning) [reply beacon_node/lighthouse_network/src/discovery/mod.rs:416]: urgh, always gets me, everytime. 

Thanks

**Review by dapplion — APPROVED** [review]
LGTM 🤝  fmt 🤝 clippy

**Comment by AgeManning** [comment]
@mergifyio queue

**Review by jxs — APPROVED** [review]
LGTM, thanks for this age!

**Comment by jxs** [comment]
@mergifyio refresh

**Comment by jxs** [comment]
@mergifyio queue

**Comment by AgeManning** [comment]
@mergifyio queue

**Comment by AgeManning** [comment]
@Mergifyio queue

**Comment by AgeManning** [comment]
@mergifyio refresh

**Comment by AgeManning** [comment]
@mergifyio refresh

**Comment by AgeManning** [comment]
@mergifyio queue

**Comment by AgeManning** [comment]
@mergifyio queue

**Pull request merged**